### PR TITLE
remove sensitive data from log messages

### PR DESCRIPTION
**Pull request by DanielW1987**

### CODE REVIEW

**Comment by DanielW1987** [comment]
[No Logging of sensitive data](https://trello.com/c/MpZAc6CH/336-no-logging-of-sensitive-data)

File: support/src/main/java/rocks/metaldetector/support/infrastructure/WithSensitiveDataRemover.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  default String removeSensitiveDataFromPayload(String originalMessage) {
+    if (originalMessage != null && originalMessage.toLowerCase().contains(PAYLOAD_IDENTIFIER)) {
+      String prePayload = originalMessage.substring(0, originalMessage.indexOf(PAYLOAD_IDENTIFIER));
+      String payload = originalMessage.substring(originalMessage.indexOf(PAYLOAD_IDENTIFIER));
+      for (String fieldName : SENSITIVE_DATA_FIELD_NAME) {
+        if (payload.contains(fieldName)) {
+          int sensitiveDataBeginIndex = payload.indexOf(fieldName) + fieldName.length() + "\":\"".length();
+          int sensitiveDataEndIndex = payload.indexOf('"', sensitiveDataBeginIndex);
```
Comment (hedgehog1833) [inline support/src/main/java/rocks/metaldetector/support/infrastructure/WithSensitiveDataRemover.java:35]: This breaks when the String contains `"` character. But that's not so easy to check... maybe checking for `","` AND `"}` would be a bit better? Still very ugly though...

Comment (DanielW1987) [reply support/src/main/java/rocks/metaldetector/support/infrastructure/WithSensitiveDataRemover.java]: I reworked it with `ObjectMapper`

File: support/src/main/java/rocks/metaldetector/support/infrastructure/WithSensitiveDataRemover.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    return headerString;
+  }
+
+  default String removeSensitiveDataFromPayload(String originalMessage) throws JsonProcessingException {
+    if (originalMessage != null && originalMessage.toLowerCase().contains(PAYLOAD_IDENTIFIER)) {
+      String prePayload = originalMessage.substring(0, originalMessage.indexOf(PAYLOAD_IDENTIFIER) + PAYLOAD_IDENTIFIER.length() - 1);
+      String payload = originalMessage.substring(originalMessage.indexOf(PAYLOAD_IDENTIFIER) + PAYLOAD_IDENTIFIER.length() - 1);
+
+      ObjectMapper mapper = new ObjectMapper();
```
Comment (hedgehog1833) [inline support/src/main/java/rocks/metaldetector/support/infrastructure/WithSensitiveDataRemover.java:39]: We have an ObjectMapper-Bean.

**Comment by hedgehog1833** [comment]
Very nice solution for that problem! 👍

File: support/src/main/java/rocks/metaldetector/support/infrastructure/WithSensitiveDataRemover.java (hunk 2 above)
Comment (DanielW1987) [reply support/src/main/java/rocks/metaldetector/support/infrastructure/WithSensitiveDataRemover.java:39]: ...but this is an interface

Comment (hedgehog1833) [reply support/src/main/java/rocks/metaldetector/support/infrastructure/WithSensitiveDataRemover.java:39]: aaaah yes :)

**Review by hedgehog1833 — APPROVED** [review]

**Pull request merged**